var/spack/repos/builtin/packages/c/package.py

```diff
@@ -14,17 +14,32 @@ class C(Package):
     homepage = "http://open-std.org/JTC1/SC22/WG14/www/standards"
     virtual = True
 
-    def test(self):
-        test_source = self.test_suite.current_test_data_dir
+    def test_c(self):
+        """Compile and run 'Hello world'"""
 
-        for test in os.listdir(test_source):
-            filepath = test_source.join(test)
-            exe_name = "%s.exe" % test
+        test_source = self.test_suite.current_test_data_dir
 
-            cc_exe = os.environ["CC"]
-            cc_opts = ["-o", exe_name, filepath]
-            compiled = self.run_test(cc_exe, options=cc_opts, installed=True)
+        cc_exe = os.environ["CC"]
+        cc_exe = which(join_path(self.prefix.bin, cc_exe))
+        if cc_exe is None:
+            raise SkipTest(f"{os.environ['CC']} not found in {self.version}")
 
-            if compiled:
-                expected = ["Hello world", "YES!"]
-                self.run_test(exe_name, expected=expected)
+        for test in os.listdir(test_source):
+            with test_part(self, f"test_c_{test}", f"Test {test}"):
+                filepath = test_source.join(test)
+                exe_name = f"{test}.exe"
+                cc_opts = ["-o", exe_name, filepath]
+                comp_exe = which(join_path(self.prefix.bin, cc_exe))
+                if comp_exe is None:
+                    raise SkipTest(f"{str(cc_exe)} not found in {self.version}")
+                compiled = cc_exe(*cc_opts)
+
+                if compiled:
+                    expected = ["Hello world", "YES!"]
+                    exe = which(join_path(self.prefix.bin, exe_name))
+                    if exe is None:
+                        raise SkipTest(f"{exe} not found in {self.version}")
```

Comment:
The above command does not install the executable (i.e., into the installation prefix); rather, it builds it in the current directory.